presto-main/src/main/java/com/facebook/presto/server/thrift/ThriftHttpResponseHandler.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.thrift;
+
+import com.facebook.airlift.http.client.HttpStatus;
+import com.facebook.airlift.http.client.thrift.ThriftResponse;
+import com.facebook.presto.server.ServiceUnavailableException;
+import com.facebook.presto.server.SimpleHttpResponseCallback;
+import com.facebook.presto.server.SimpleHttpResponseHandlerStats;
+import com.facebook.presto.spi.ErrorCodeSupplier;
+import com.facebook.presto.spi.PrestoException;
+import com.google.common.util.concurrent.FutureCallback;
+
+import java.net.URI;
+
+import static com.facebook.airlift.http.client.HttpStatus.OK;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class ThriftHttpResponseHandler<T>
```

Comment:
I assume that this is just an almost verbatim copy of the SimpleHttpResponseHandler. I wonder if the `SimpleHttpResponseHandler` can be reused to avoid copy-paste?

Reply:
SimpleHttpResponseHandler<T>
        implements FutureCallback<BaseResponse<T>>
while
public class ThriftHttpResponseHandler<T>
        implements FutureCallback<ThriftResponse<T>>

issue is that ThriftResponse doesn't implement com.facebook.presto.server.smile.BaseResponse which sits in presto-main .

We can maybe have a follow up task to move BaseResponse upto airlift and then make ThriftResponse extend from BaseResponse to have similar handling.